Sources/OpenAPIRuntime/Conversion/CurrencyExtensions.swift

```diff
@@ -320,23 +383,49 @@ extension Converter {
 
     func getOptionalQueryItems<T>(
         in queryParameters: [URLQueryItem],
+        style: ParameterStyle?,
+        explode: Bool?,
         name: String,
         as type: [T].Type,
         convert: (String) throws -> T
     ) throws -> [T]? {
-        let untypedValues = queryParameters.filter { $0.name == name }
-        return try untypedValues.map { try convert($0.value ?? "") }
+        let (_, resolvedExplode) = try ParameterStyle.resolvedQueryStyleAndExplode(
+            name: name,
+            style: style,
+            explode: explode
+        )
+        let untypedValues =
+            queryParameters
+            .filter { $0.name == name }
+            .map { $0.value ?? "" }
+        // If explode is false, some of the items might have multiple
+        // comma-separate values, so we need to split them here.
+        let processedValues: [String]
+        if resolvedExplode {
+            processedValues = untypedValues
+        } else {
+            processedValues = untypedValues.flatMap { multiValue in
+                multiValue
+                    .split(separator: ",", omittingEmptySubsequences: false)
```

Comment:
I'm sorry for the late comment, but in theory I assume it would be possible to have values with commas in them. Would the commas be escaped or percent-encoded in this case? Just want to make sure this split doesn't split the string at incorrect places.

Reply:
Yeah exactly, if the comma is part of the value, it is percent escaped.